src/parser.cpp

```diff
@@ -50,8 +50,8 @@ namespace ada::parser {
     // Optimization opportunity. Most websites do not have fragment.
     std::optional<std::string_view> fragment = helpers::prune_fragment(url_data);
     if(fragment.has_value()) {
-      url.fragment = unicode::percent_encode(*fragment,
-                                             ada::character_sets::FRAGMENT_PERCENT_ENCODE);
+      url.set_fragment(unicode::percent_encode(*fragment,
```

Comment:
I think ada::parser should never call `url-setters` and `url-getters` since it will be implemented differently between `href_aggregator` and `ada::url`